### PR TITLE
Ensure JAXB and javax.activation are not required for JAX-RS

### PR DESCRIPTION
**Pull request by rmannibucau**
Goal is to ensure default java11 experience is still good when not needing JAXB/activation (which is pretty much often these days with JAX-RS)

### CODE REVIEW

File: core/src/main/java/org/apache/cxf/service/model/EndpointInfo.java [hunk 1]
```diff
@@ -94,7 +95,12 @@ public String getAddress() {
 
     public void setAddress(String addr) {
         if (null == address) {
-            address = EndpointReferenceUtils.getEndpointReference(addr);
+            // address = EndpointReferenceUtils.getEndpointReference(addr); loads jaxb and we want to avoid it
+            final EndpointReferenceType reference = new EndpointReferenceType();
+            final AttributedURIType a = new AttributedURIType();
+            a.setValue("/");
```
Comment (coheigea) [inline core/src/main/java/org/apache/cxf/service/model/EndpointInfo.java]: This should be: a.setValue(address);

Comment (rmannibucau) [reply core/src/main/java/org/apache/cxf/service/model/EndpointInfo.java]: Hmm, sure? address is null here so it defaults to "/" but good catch i forgot the else ;)

**Comment by coheigea** [comment]
ProviderFactory has a method:

 static int comparePriorityStatus(Class<?> cl1, Class<?> cl2) {
        Integer value1 = AnnotationUtils.getBindingPriority(cl1);
        Integer value2 = AnnotationUtils.getBindingPriority(cl2);
        return value1.compareTo(value2);
    }

AnnotationUtils.has: import javax.annotation.Priority;

So I guess AnnotationUtils.getBindingPriority needs to use some reflection as well here.

**Comment by rmannibucau** [comment]
@coheigea have to admit i used CDI to test so commons-annotation was there (and @Priority) but good catch, updated

**Comment by rmannibucau** [comment]
@coheigea fixed it, misread the getClassAnnotation method (thought the first test was on the annotation but it was on the class). Thanks for catching it.

**Comment by coheigea** [comment]
Thanks @rmannibucau

**Pull request merged**